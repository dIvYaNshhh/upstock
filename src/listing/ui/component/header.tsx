import React from 'react'
import { View, Text } from 'react-native'
import styled from 'styled-components'

type Props = {
  title: string
}
export const Header = ({ title }: Props) => {
  return (
    <Container>
      <Title>{title}</Title>
    </Container>
  )
}

const Container = styled(View)`
  background-color: #7a007a;
`

const Title = styled(Text)`
  color: #fff;
  font-size: 20px;
  padding: 16px;
`
