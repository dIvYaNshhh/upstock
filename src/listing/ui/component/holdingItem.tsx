import React, { useState } from 'react'
import { Text, View, TouchableOpacity } from 'react-native'
import styled from 'styled-components'
import Collapsible from 'react-native-collapsible'
import { HoldingItem } from 'src/listing/gateway/types'
import { usePortfolioData } from 'src/listing/bussinessLogic/usePortfolioData'

interface Props {
  item: HoldingItem
}
const currencySymbol = 'â‚¹'
export const HoldingListItem = React.memo(({ item }: Props) => {
  const { currentValue, investmentValue, pnl, todaysPnl } = usePortfolioData({
    item,
  })

  const [collapsed, setCollapsed] = useState(true)

  const toggleExpanded = () => {
    setCollapsed(!collapsed)
  }

  return (
    <View>
      <Container onPress={toggleExpanded}>
        <Column flexEnd={false}>
          <Symbol>{item.symbol}</Symbol>
          <TextView>{item.quantity}</TextView>
        </Column>

        <Column flexEnd={true}>
          <TextView>
            {'LTP: '}
            <MildBoldText>{`${currencySymbol} ${item.ltp}`}</MildBoldText>
          </TextView>
          <TextView>
            {'P/L: '}
            <MildBoldText>{`${currencySymbol} ${todaysPnl.toFixed(
              2
            )}`}</MildBoldText>
          </TextView>
        </Column>
      </Container>
      <Collapsible collapsed={collapsed}>
        <ExpandedSection>
          <Row>
            <MildBoldText>{'Current Value: '}</MildBoldText>
            <TextView>{`${currencySymbol} ${currentValue.toFixed(
              2
            )}`}</TextView>
          </Row>
          <Row>
            <MildBoldText>{'Total Investment: '}</MildBoldText>
            <TextView>{`${currencySymbol} ${investmentValue.toFixed(
              2
            )}`}</TextView>
          </Row>
          <Row>
            <MildBoldText>{`Today's Profit & Loss: `}</MildBoldText>
            <TextView>{`${currencySymbol} ${pnl.toFixed(2)}`}</TextView>
          </Row>
          <Space />
          <Row>
            <MildBoldText>{'Profit & Loss: '}</MildBoldText>
            <TextView>{`${currencySymbol} ${pnl.toFixed(2)}`}</TextView>
          </Row>
        </ExpandedSection>
      </Collapsible>
    </View>
  )
})

const Container = styled(TouchableOpacity)`
  padding: 16px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`
const Column = styled(View)<{
  flexEnd: boolean
}>`
  flex: 1;
  flex-direction: column;
  align-items: ${({ flexEnd }) => (flexEnd ? 'flex-end' : 'flex-start')};
`

const Symbol = styled(Text)`
  font-weight: bold;
  font-size: 16px;
  padding: 2px;
`
const TextView = styled(Text)`
  font-size: 16px;
  padding: 2px;
`

const MildBoldText = styled(Text)`
  font-weight: bold;
  font-size: 16px;
  padding: 2px;
`
const ExpandedSection = styled(View)`
  padding: 16px;
`
const Row = styled(View)`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`

const Space = styled(View)`
  height: 16px;
`
