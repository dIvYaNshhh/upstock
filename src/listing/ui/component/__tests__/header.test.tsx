import React from 'react'
import { render, RenderAPI } from 'react-native-testing-library'
import { Header } from 'src/listing/ui/component/header'

describe('Component - Header', () => {
  let instance: RenderAPI

  beforeEach(() => {
    instance = render(<Header title="Upstox Holding" />)
  })

  it('matches the snapshot', () => {
    expect(instance.toJSON()).toMatchSnapshot()
  })
})
