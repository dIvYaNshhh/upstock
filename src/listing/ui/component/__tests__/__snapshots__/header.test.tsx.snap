// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Header matches the snapshot 1`] = `
<View
  className="sc-bdVaJa bVqVlE"
>
  <Text
    className="sc-bwzfXH fwuZZg"
  >
    Upstox Holding
  </Text>
</View>
`;
